docs/howto/enumerations.md

```diff
@@ -0,0 +1,100 @@
+
+Although often being of `dataType` [`sc:Text`](http://schema.org/Text), enumerations can also be of other `dataType`s (e.g. `sc:Integer`).
+
+For example, the titanic dataset uses an enumeration to specify
+that the passengers gender field, of type `sc:Text`, can only take two values: `male` or `female`.
+
+## Declaring enumerations
+
+There are two ways to declare enumerations:
+
+1. Directly on a `ml:Field`. This is a simple way to define an enumeration
+   but has some limitations discussed below.
+2. Using a dedicated `ml:RecordSet`, holding the enumeration values. Such
+   definition allows enumerations to be fully defined, with a name and a
+   semantic URL.
+
+### `ml:Field` enumeration
+
+In the following example, extracted from the [coco2014 Croissant definition](https://github.com/mlcommons/croissant/blob/main/datasets/coco2014/metadata.json), the field `supercategory` is declared to be
+an enumeration using the `"ml:isEnumeration": true` property. This means that
+there is a finite number of `supercategory` values, and that it is expected
+that many rows will have the same value.
+
+https://github.com/mlcommons/croissant/blob/6c8eca2925191577a7101dcabd0a880aca03cae3/datasets/coco2014/metadata.json#L259-L271
+
+The field is of `dataType` `sc:Text`, but also has the type `sc:name`. This
+means that `supercategory` is expected to be a text, and that this text is
+the name of the enumeration value   ` naming itself (the
+supercategory), makes sense to humans, and could therefore be used as it in
+UIs.
+
+Although being straightforward, this approach of defining an enumeration has several limitations:
+
+1. The only way to know the cardinality of the enumeration (the number of
+   distinct values it can take), is to read the full `RecordSet` in which
+   the enumeration is defined, or to have data format native support for
+   enumerations.
+2. It doesn't allow specifying an enumeration value, a name and a semantic URL
+   together. If the underlying format allows it (e.g. Parquet), one might
+   declare an enumeration value (0, 1, 2, etc.) and a name, but no semantic
+   URL.
+
+For the above reasons, it is recommended to use a `RecordSet` to define an
+enumeration unless:
+
+1. the data format used by to store the `RecordSet` natively supports the
+  concept of enumerations, which should make it fast to list all possible
+  enumerations.\
+  or
+2. the `RecordSet` in which the enumeration is defined is known to be small.
+  This is the case in the example above, as the parent `RecordSet` is being
+  used to define another enumeration.
+
+### `ml:RecordSet` enumeration
+
+In the following example, also extracted from the [coco2014 Croissant definition](https://github.com/mlcommons/croissant/blob/main/datasets/coco2014/metadata.json), the `categories` `ml:RecordSet` is declared as being an enumeration:
```

Comment:
I got really confused by this example, because I expected it to match the one above. It took me a while to understand that this was for "category" and not "supercategory" like the one above.